zeton/api/user.py

```diff
@@ -0,0 +1,37 @@
+from flask import request, url_for, g
+from werkzeug.security import generate_password_hash
+from werkzeug.utils import redirect
+
+from zeton.api import bp
+from zeton.data_access import users
+from zeton.data_access.bans import insert_all_default_bans
+from zeton.data_access.users import add_new_user, get_child_id, associate_child_with_caregiver
+
+
+@bp.route("/user", methods=['POST'])
+def register():
+    users.load_logged_in_user_data()
+    username = request.form['username']
+    password = request.form['password']
+    password_hash = generate_password_hash(password)
+    role = request.form.get('role') or 'caregiver'
+    firstname = request.form.get('name') or '-'
```

Comment:
To szczegół, ale wolę dopytać. Mogę dodać dziecko jakby bez imienia tylko ze znakiem '-' ? nieładnie to wygląda. Plus może chciałbyś dodać do tego capitalize() ?

Reply:
Kombinuję z uproszczeniem rejestracji do podania samego Username/Emaila i hasła, ale póki co to powoduje więcej problemów niż pożytku, więc albo odpuszczę albo wymyślę jak to zrobić lepiej. Nie zwróciłem uwagi, że imię jest jednak wykorzystywane praktycznie wszędzie. Dzięki za uwagę, gdzieś w ferworze walki z tabelami mi to umknęło.

Co do capitalize to nie wiem, czy jestem zwolennikiem poprawiania inputu użytkownika. A co jeśli ktoś sobie zażyczy jako imię wpisać: "bOmBeLek" ? ;)

Reply:
Pole `firstname` w bazie nie jest wymagane. Podczas rejestracji nie wymagałbym uzupełnienia tego pola. W momencie gdy user nie poda żadnego imienia wpisujemy do bazy `NONAME`. Natomiast gdy użytkownik poda jakiekolwiek znaki wówczas sprawdzamy, aby były minum 3 liter alfabetu (obojętnie czy małe czy tez wielkie litery) - walidacja na poziomie frontu. Jako że chcemy uprościć rejestrację proponuję walidację pola `imię` przenieś do nowego issue. Na tą chwilę zakładamy, że użytkownik wpisał poprawne imię lub nie wpisał wcale. Bedzie mógł zmienić imię w ustaiweniach. Też będzie na to osobne zadanie.

Reply:
Wyobraziłam sobie jak elegancko może się prezentować aplikacja i nie pomyślałam o  "bOmBeLeK" :D